frontend/src/components/Nav.tsx

```diff
@@ -16,9 +20,15 @@ export default function Nav() {
                         </Link>
                     </li>
                     <li>
-                        <Link to={PathConstants.LOGIN} className="text-white py-2 px-4 rounded-lg bg-red-500 mr-2">
-                            Login
-                        </Link>
+                        {user ? (
+                            <Link to={PathConstants.LOGOUT} className="text-white py-2 px-4 rounded-lg bg-red-500 mr-2">
```

Comment:
Change this to button a and make a function to Logout and set is as onClick event for the button.. make sure to call the actual API call (api/user/logout) to remove sessionid before setting user to undefined